CMakeLists.txt

```diff
@@ -7,6 +7,27 @@ if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
     SET(CMAKE_INSTALL_PREFIX "/usr" CACHE PATH "..." FORCE)
 endif(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
 
+# Un-prefix absolute install dirs
+
+function(unprefix VAR)
+    string(FIND "${${VAR}}" "${CMAKE_INSTALL_PREFIX}" POS)
+    if("${POS}" EQUAL 0)
+        string(LENGTH "${CMAKE_INSTALL_PREFIX}" POS)
+        MATH(EXPR POS "${POS}+1")
+        string(SUBSTRING "${${VAR}}" "${POS}" -1 ${VAR})
+        SET(${VAR} ${${VAR}} PARENT_SCOPE)
+    endif()
+endfunction()
+
+unprefix(CMAKE_INSTALL_BINDIR)
+unprefix(CMAKE_INSTALL_LIBEXECDIR)
+unprefix(CMAKE_INSTALL_LIBDIR)
+unprefix(CMAKE_INSTALL_INCLUDEDIR)
+unprefix(CMAKE_INSTALL_DATADIR)
+unprefix(CMAKE_INSTALL_LOCALEDIR)
+unprefix(CMAKE_INSTALL_MANDIR)
+unprefix(CMAKE_INSTALL_DOCDIR)
```

Comment:
Why undo stuff that is actually seemingly (haven't tested, just read on the web) fine?

Simply don't hard-coded-ly prepent CMAKE_INSTALL_PREFIX and then we should be done.

Reply:
CMake does **NOT** prepend `CMAKE_INSTALL_PREFIX` to `CMAKE_INSTALL_*DIR` variables (tested it). The expected behaviour is to provide an absolute path to `CMAKE_INSTALL_PREFIX`, and relative to that, all the `CMAKE_INSTALL_*DIR`. It is not an error to use absolute paths, but it should be then handled somehow. 

This is not so much a hack, but a way to accommodate that 1% of distros that do this. Normally, we would depend on `CMAKE_INSTALL_FULL__*DIR`, but what if they override that as well?

http://cmake.3232098.n2.nabble.com/CMAKE-INSTALL-LIBDIR-full-path-or-relative-td7302556.html
https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html

There are two solutions:
1. Use this sanity-fix in every project
2. Use `CMAKE_INSTALL_FULL__*DIR` in install funtions and hope for the best

Circle the desired option.

Reply:
What I normally see, and I mixed it up, is CMAKE_INSTALL_FULL_* in install target rules. So circle is made around No. 2.

Reply:
> What I normally see, and I mixed it up, is CMAKE_INSTALL_FULL_* in install target rules. So circle is made around No. 2.

Done.